### PR TITLE
Decouple storages

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️/❌
| New feature?  | ✔️/❌
| Breaks BC?    | ✔️/❌
| Fixed issues  |

### CODE REVIEW

File: src/DbSchemaManager.php [hunk 1]
```diff
@@ -191,51 +194,51 @@ public function ensureTables(): void
...
     /**
      * Gets name of the table for storing RBAC items (roles and permissions).
      *
-     * @return string Table name
+     * @return string|null Table name.
      *
      * @see $itemsTable
      */
-    public function getItemsTable(): string
+    public function getItemsTable(): ?string
```
Comment (vjik) [inline src/DbSchemaManager.php:217]: To this change need adapt `RbacCycleInit`. May be add tests for `RbacCycleInit` (sugggest make this in separate PR) ?

File: src/ItemsStorage.php [hunk 2]
```diff
@@ -28,6 +28,7 @@
  *     createdAt: int|string,
  *     updatedAt: int|string
  * }
+ * @psalm-type ItemsIndexedByName = array<string, Permission|Role>
```
Comment (vjik) [inline src/ItemsStorage.php:31]: Move it and types in `getChildren` and other methods to `yiisoft/rbac`

File: README.md [hunk 3]
```diff
@@ -125,15 +126,22 @@ $assignmentsStorage = new AssignmentsStorage(
...
+        // Requires https://github.com/yiisoft/rbac-rules-container or other compatible factory.
+        ruleFactory: $rulesContainer,
+    ),
 );
 $manager->addPermission(new Permission('posts.create'));
 ```
 
+> Note wrapping manager with decorator - it additionally provides database transactions to guarantee data integrity:
+
+> Note that it's not necessary to use both DB storages. Combining different implementations is possible. A quite popular
```
Comment (vjik) [inline README.md:142]: It should be mentioned here that if used different implementations, then need use `Manager` without decorator.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 - New #23: Add `DbSchemaManager` and SQL dumps for working with schema (@arogachev)
 - Chg #23: Remove CLI dependencies (@arogachev)
+- Enh #60: Decouple storages (@arogachev)
```
Comment (vjik) [inline CHANGELOG.md:7]:
```suggestion
- Enh #60: Make database tables independent (remove foreign keys) (@arogachev)
- New #60: Add `TransactionalManagerDecorator` (@arogachev)
```

**Review by vjik — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (arogachev) [reply CHANGELOG.md:7]: It's not about database tables and foreign keys only.

Comment (vjik) [reply CHANGELOG.md:7]: Please, add what I missed.

File: src/ItemsStorage.php (hunk 2 above)
Comment (arogachev) [reply src/ItemsStorage.php:31]: https://github.com/yiisoft/rbac/pull/167

File: src/DbSchemaManager.php (hunk 1 above)
Comment (arogachev) [reply src/DbSchemaManager.php:217]: OK. Let's do it separately - #63.

File: README.md (hunk 3 above)
Comment (arogachev) [reply README.md:142]: Reverted because when using at least 1 DB storage it's always needed.

**Review by vjik — APPROVED** [review]

File: composer.json [hunk 5]
```diff
@@ -19,17 +19,20 @@
         "chat": "https://t.me/yii3en",
         "source": "https://github.com/yiisoft/rbac-cycle-db"
     },
+    "minimum-stability": "dev",
     "require": {
         "php": "^8.0",
         "cycle/database": "^2.1",
-        "yiisoft/rbac": "^1.0.2"
+        "yiisoft/rbac": "dev-master"
```
[the remark this answers is not in the document]
Comment (arogachev) [reply composer.json:26]: #64.
